compose-lint-checks/src/test/java/slack/lint/compose/PreviewPublicDetectorTest.kt

```diff
@@ -30,6 +31,7 @@ class PreviewPublicDetectorTest : BaseSlackLintTest() {
     lint().files(kotlin(code)).allowCompilationErrors().run().expectClean()
   }
 
+  @Ignore("TODO code review")
```

Comment:
@mrmans0n @chrisbanes I was surprised to find this test that starts failing after the fix. The test seems to disagree with the check's stated purpose, do you have more details?

<img width="758" alt="image" src="https://github.com/slackhq/compose-lints/assets/1361086/3be767e4-b6a1-472a-b644-4185d86dea68">

Reply:
Not sure. I assume the second function declaration is a typo

Reply:
Tried to remember why this was done... but I can't recall exactly. The idea I think was around previews being alright being public, private or whatever unless they had an annotated param (which made them stupid to be public then). But the meaning of the rule changed over time (adding a param to control this was added, many behavior changes happened later).

FWIW this was simplified to just check for public + preview, and ignore everything else in my repo https://github.com/mrmans0n/compose-rules/blob/main/rules/common/src/main/kotlin/io/nlopez/compose/rules/PreviewPublic.kt

You could do the same and call it a day.

Reply:
[cd47d5f](https://github.com/slackhq/compose-lints/pull/206/commits/cd47d5fd50fdbadbb6302112c72f8296d7e3e275)